includes/fbproduct.php

```diff
@@ -376,13 +376,13 @@ public function add_sale_price( $product_data, $for_items_batch = false ) {
 			intval( round( $this->get_price_plus_tax( $sale_price ) * 100 ) );
 
 			$sale_start =
-			( $date     = get_post_meta( $this->id, '_sale_price_dates_from', true ) )
-			? date_i18n( 'Y-m-d', $date ) . self::MIN_TIME
+			( $date     = $this->woo_product->get_date_on_sale_from() )
+			? date_i18n( 'Y-m-d', $date->getOffsetTimestamp() ) . self::MIN_TIME
 			: self::MIN_DATE_1 . self::MIN_TIME;
 
 			$sale_end =
-			( $date   = get_post_meta( $this->id, '_sale_price_dates_to', true ) )
-			? date_i18n( 'Y-m-d', $date ) . self::MAX_TIME
+			( $date   = $this->woo_product->get_date_on_sale_to() )
+			? date_i18n( 'Y-m-d', $date->getOffsetTimestamp() ) . self::MAX_TIME
```

Comment:
```suggestion
			? date_i18n( DATE_ATOM, $date->getOffsetTimestamp() )
```

or 

```suggestion
			? date_i18n( 'Y-m-d\TH:i:s+00:00', $date->getTimestamp() )
```

Reply:
Thanks, @message-dimke, you're assessment is 100% correct. I have applied your suggestion which works well. 

Thanks for taking the time to look at this. And I learned about the DATE_ATOM constant which is a bonus :-)